### PR TITLE
Dihuang@tusimple

### PR DESCRIPTION
**Pull request by seusofthd**
@tqchen Fix the bugs for Ctrl+C signal as you mentioned. Using SignalHandler will report several warnings at the compile time which is caused by the package itself. Just ignore it.

### CODE REVIEW

**Comment by tqchen** [comment]
please do a squash and I will merge it in. Thanks!

File: tracker/yarn/src/main/java/org/apache/hadoop/yarn/dmlc/ApplicationMaster.java [hunk 1]
```diff
@@ -601,7 +601,7 @@ private synchronized void onContainersCompleted(List<ContainerStatus> status) {
                         return;
                     }
                 } catch (Exception e) {
-                    LOG.warn(e.getMessage());
+                        LOG.warn(e.getMessage());
```
Comment (tqchen) [inline tracker/yarn/src/main/java/org/apache/hadoop/yarn/dmlc/ApplicationMaster.java]: seems to be extra space or tab?

File: tracker/yarn/src/main/java/org/apache/hadoop/yarn/dmlc/Client.java [hunk 2]
```diff
@@ -34,6 +34,9 @@
 import org.apache.hadoop.yarn.util.ConverterUtils;
 import org.apache.hadoop.yarn.util.Records;
 
+import sun.misc.Signal;
```
Comment (tqchen) [inline tracker/yarn/src/main/java/org/apache/hadoop/yarn/dmlc/Client.java]: does this mean the signal handler is sun JVM dependent?

**Comment by seusofthd** [comment]
yes. On my machine I just use openjdk. It can compile.

**Comment by tqchen** [comment]
see https://xgboost.readthedocs.org/en/latest/dev-guide/contribute.html#how-to-combine-multiple-commits-into-one

**Comment by tqchen** [comment]
@seusofthd can you merge your commits? Thanks

**Comment by seusofthd** [comment]
Oh... Ok, I forgot to merge. Let me merge it when I come to office tmr

**Comment by seusofthd** [comment]
@tqchen I followed the instructions on the tutorial and solved all the conflicts, but it seems that it did not work...

**Comment by tqchen** [comment]
could due to there is a merge back operation in your log. One thing you can do is to backup your code, reset to a old version and copy it back. but this can be quite dangerous so be careful to backup first

**Comment by seusofthd** [comment]
ok...

**Comment by tqchen** [comment]
i.e. start from a clean fork Thanks for all the effort

**Comment by seusofthd** [comment]
This would be a easier way i think. I will start from a clean fork then.

**Comment by tqchen** [comment]
any update on this?

**Comment by seusofthd** [comment]
@tqchen the system now runs smoothly in our cluster. I haven't find any other problems yet. Maybe we can write the label scheduling response in appmaster when cdh added label scheduling in their latest version

**Comment by tqchen** [comment]
I mean, did you ever PRed this from another fork?

**Comment by seusofthd** [comment]
@tqchen I remember I've already done a clean fork before and you've merged it

**Pull request closed without merging**